### PR TITLE
Update doc guidelines regarding footnote usage

### PR DESCRIPTION
**Pull request by jboxman**
@vikram-redhat, FYI.

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@openshift/team-documentation @openshift/team-aligned-docs

**Comment by mburke5678** [comment]
Can we get a real-world example of each method? Or, at least, a links to modules that use the methods (more dangerous)?

**Comment by jboxman** [comment]
@mburke5678, the only method we actually use anywhere is the deprecated `footnoteref` syntax. The other styling approach is new.

File: contributing_to_docs/doc_guidelines.adoc [hunk 1]
```diff
@@ -688,6 +688,35 @@ some code block
...
+:note-text: This is a footnote
+
+footnote:[{note-text}]
+....
+
+Avoid using `footnoteref`.
+
+[IMPORTANT]
+====
+The `footnoteref` direct is deprecated in asciidoctor and causes a build warning when `ascii_binder` is run.
```
Comment (lamek) [inline contributing_to_docs/doc_guidelines.adoc:712]: Should this be "directive"?

**Comment by lamek** [comment]
LGTM - one small q.

File: contributing_to_docs/doc_guidelines.adoc (hunk 1 above)
Comment (jboxman) [reply contributing_to_docs/doc_guidelines.adoc:712]: @lamek, ha, yes!

**Pull request merged**